src/timer.rs

```diff
@@ -350,10 +350,10 @@ macro_rules! hal {
                 }
 
                 fn set_timeout_ticks(&mut self, ticks: u32) {
-                    let psc = u16((ticks - 1) / (1 << 16)).unwrap();
+                    let psc = u16(ticks / (1 << 16)).unwrap();
                     self.tim.psc.write(|w| w.psc().bits(psc));
 
-                    let arr = u16(ticks / u32(psc + 1)).unwrap();
```

Comment:
I'm fairly sure the only required change here is `u16(ticks / psc as u32 + 1).unwrap()`. I don't see how removing the `ticks - 1` changes anything here?

Reply:
Ok, let's do the calculations manually. I agree that this code is non-obvious and I could add some comments.
```
ticks = 65536
psc = (ticks - 1) / 65536 = 65535 / 65536 = 0
arr = 65536 / (psc + 1) = 65536 / (0 + 1) = 65536 / 1 = 65536 = overflow
```

With the -1 removed:
```
ticks = 65536
psc = ticks / 65536 = 65536 / 65536 = 1
arr = 65536 / (psc + 1) = 65536 / (1 + 1) = 65536 / 2 = 32768
```
The result is correct because  `32768 * (psc + 1) = ticks` is true.